packages/material-ui/src/Grid/Grid.d.ts

```diff
@@ -78,18 +78,88 @@ export type GridClassKey =
   | 'grid-xs-12';
 
 export interface GridTypeMap<P = {}, D extends React.ElementType = 'div'> {
-  props: P &
-    Partial<Record<Breakpoint, boolean | GridSize>> & {
-      alignContent?: GridContentAlignment;
-      alignItems?: GridItemsAlignment;
-      container?: boolean;
-      direction?: GridDirection;
-      item?: boolean;
-      justify?: GridJustification;
-      spacing?: GridSpacing;
-      wrap?: GridWrap;
-      zeroMinWidth?: boolean;
-    };
+  props: P & {
+    /**
+     * Defines the `align-content` style property.
+     * It's applied for all screen sizes.
+     */
+    alignContent?: GridContentAlignment;
+    /**
+     * Defines the `align-items` style property.
+     * It's applied for all screen sizes.
+     */
+    alignItems?: GridItemsAlignment;
+    /**
+     * The content of the component.
+     */
+    children?: React.ReactNode;
+    /**
+     * If `true`, the component will have the flex *container* behavior.
+     * You should be wrapping *items* with a *container*.
+     */
+    container?: boolean;
+    /**
+     * Defines the `flex-direction` style property.
+     * It is applied for all screen sizes.
+     */
+    direction?: GridDirection;
+    /**
+     * If `true`, the component will have the flex *item* behavior.
+     * You should be wrapping *items* with a *container*.
+     */
+    item?: boolean;
+    /**
+     * Defines the `justify-content` style property.
+     * It is applied for all screen sizes.
+     * @deprecated Use `justifyContent` instead, the prop was renamed
+     */
+    justify?: GridJustification;
+    /**
+     * Defines the `justify-content` style property.
+     * It is applied for all screen sizes.
+     */
+    justifyContent?: GridJustification;
+    /**
+     * Defines the number of grids the component is going to use.
+     * It's applied for the `lg` breakpoint and wider screens if not overridden.
+     */
+    lg?: boolean | GridSize;
+    /**
+     * Defines the number of grids the component is going to use.
+     * It's applied for the `md` breakpoint and wider screens if not overridden.
+     */
+    md?: boolean | GridSize;
+    /**
+     * Defines the number of grids the component is going to use.
+     * It's applied for the `sm` breakpoint and wider screens if not overridden.
+     */
+    sm?: boolean | GridSize;
```

Comment:
These props did not exist before,  I believe we should drop them.

Reply:
These sm, md, etc. props were coming from 

https://github.com/mui-org/material-ui/blob/7fe8a4254155c5475c3e14971b0c01d88f57efa8/packages/material-ui/src/Grid/Grid.d.ts#L82

We made them explicit. It's coming from the on v5 that I cherry-picked.